pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -169,8 +174,91 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		#[pallet::call_index(0)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn create_pool(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn create_pool(
+			origin: OriginFor<T>,
+			curve: CurveInput<CurveParameterInputOf<T>>,
+			currencies: BoundedVec<TokenMetaOf<T>, T::MaxCurrencies>,
+			state: PoolStatus<Locks>,
+			denomination: u8,
+			pool_manager: AccountIdOf<T>,
+			transferable: bool,
+			team: Team<AccountIdOf<T>>,
+		) -> DispatchResult {
+			let who = T::PoolCreateOrigin::ensure_origin(origin)?;
+
+			let currency_length = currencies.len();
+
+			ensure!(
+				(1..=(T::MaxCurrencies::get()).saturated_into()).contains(&currency_length),
+				Error::<T>::CurrenciesNumber
+			);
+
+			let checked_curve = curve.try_into().map_err(|_| Error::<T>::InvalidCoefficients)?;
```

Comment:
This operation can still fail. Depending on the selected `CurveparameterType`, the int bits may not fit. The error from `try_into` is ignored. I think it doesn't really matter to know which parameter is wrong.

Reply:
I would rename the error `InvalidInput`, since as far as this pallet is concerned, it does not know exactly what in the conversion failed: it just knows that it could not convert the input to the expected type.

Reply:
renamed the error to: [633e119](https://github.com/KILTprotocol/kilt-node/pull/768/commits/633e119696a15443164ebc652088950e15e3b047)